manufacturing-client/src/main.rs

```diff
@@ -937,3 +959,52 @@ impl KeyReference {
         }
     }
 }
+
+const IPV6_DEFAULT: &str = "00000000000000000000000000000000"; //DevSkim: ignore DS173237
+const IPV4_DEFAULT: &str = "00000000";
+
+fn get_default_network_iface() -> Result<Option<String>> {
```

Comment:
Why does this return a `Result<_>` if it can't fail?
I'm wondering if the reading of the route tables should return errors if they error out?

Reply:
it returns either the mac address string or None , if None there was some error determining the interface.
Are you suggesting instead of None, the function should send error ? (for more details!)

Reply:
I think that since, for instance, we are not propagating the errors of `std::fs::File::open`, it makes no sense to return `Result`, since there won't be any errors.

If we are going to return `Some` or `None` (like we are doing now), then it would make more sense to return an `Option` instead, or to continue returning `Result`, but propagating the errors that we could encounter.

Reply:
handled by propagating std error for more visibility on error msg